### DISCUSSION TITLE
Update agent description and system message based on groupchat context

### DISCUSSION THREAD

**Discussion by itskunalsamanta** [microsoft/autogen#4361] [Q&A]
I am trying to simulate conversations among multiple agents using groupchat. The groupchat manager in "auto" mode selects the person to speak based on the agent description, which is static. 
What is the best way to have the agent descriptions and system message change for a particular agent, based on chat context after each message is broadcasted by the groupchat manager?

**Comment by ekzhu** [comment]
You would need to do this inside the group chat manager to determine if the description should be updated.

In v0.4 agentchat, this would be a custom team implementing the `BaseGroupChat`.

**Reply by itskunalsamanta** [reply]
I am working with v0.2, is there a straightforward way to do it there, or do I have to modify the group chat manager code with my own custom code?

**Reply by ekzhu** [reply]
There isn't a straightforward way to do this. I suggest looking into custom speaker selection function for group chat. You can incorporate a stateful selection method there.
